File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {

    public double productionRatePerHour(int speed) {
        var productionRate = speed * 221;
        var successRate = 0.0;
        if (speed <= 4) {
            successRate = 1;
        } else if (speed <= 8) {
            successRate = 0.9;
        } else if (speed == 9) {
            successRate = 0.8;
        } else {
            successRate = 0.77;
        }
        
        return productionRate * successRate;
    }

    public int workingItemsPerMinute(int speed) {
        return (int) Math.floor(productionRatePerHour(speed) / 60);
    }
}

File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
using System;

static class AssemblyLine
{
    private const int CARS_PER_HOUR = 221;

    public static double SuccessRate(int speed)
    {
        return speed switch
        {
            0    => 0,
            <= 4 => 1.0,
            <= 8 => 0.9,
            <= 9 => 0.8,
            10   => 0.77,
            _    => throw new ArgumentOutOfRangeException("speed range is 0 to 10. supplied " + speed),
        };
    }
    
    public static double ProductionRatePerHour(int speed)
    {
        return speed * CARS_PER_HOUR * SuccessRate(speed);
    }

    public static int WorkingItemsPerMinute(int speed)
    {
        return (int) Math.Floor(ProductionRatePerHour(speed) / 60);
    }
}
